### ISSUE TITLE
First class heap allocations from different threads

### ISSUE THREAD

**Issue by robertblaketaylor** [microsoft/mimalloc#925]
I was interested in using the direct heap functions for tracking heaps directly in our codebase, but see the caveat that:

```
but can only allocate from the same thread that created it.
```

Do you have any plans on removing this restriction?

**Comment by daanx** [comment]
Yes, I am planning to add this but it is not entirely trivial to do in a performant way. Currently working on improving behavior with many threads being created/terminated and blocked threads but I hopefully get to fc heaps after that.

**Comment by 1027663760** [comment]
https://github.com/microsoft/mimalloc/issues/720

**Comment by robertblaketaylor** [comment]
Any updates on this?
